cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/common/tools/ResourcesPluginUtil.java

```diff
@@ -45,15 +72,76 @@ public static void refreshConfigurationProject() {
     public static IProject getGeneratorConfigurationProject()
         throws GeneratorProjectNotExistentException, CoreException {
 
-        IProject generatorProj =
-            ResourcesPlugin.getWorkspace().getRoot().getProject(ResourceConstants.CONFIG_PROJECT_NAME);
+        generatorProj = ResourcesPlugin.getWorkspace().getRoot().getProject(ResourceConstants.CONFIG_PROJECT_NAME);       
         if (!generatorProj.exists()) {
-            throw new GeneratorProjectNotExistentException();
+        	          
+                MessageDialog dialog = new MessageDialog(Display.getDefault().getActiveShell(),
+                    "Generator configuration project not found!", null,
+                    "Cobigen_templates folder is not imported. Do you want to download latest templates and use it", 0,
+                    new String[] { "Update", "Cancel" }, 1);
+                dialog.setBlockOnOpen(true);
+                dialog.open();
+              
+            /*if (result == 0) {
+                //isUpdateDialogShown = false;
+                return null;
+
+            } else {
+                MessageDialog.openError(Display.getDefault().getActiveShell(),
+                    "Generator configuration project not found!",
+                    "The project '" + ResourceConstants.CONFIG_PROJECT_NAME
+                        + "' containing the configuration and templates is currently not existent. Please create one or check it out from SVN as stated in the user documentation.");
+
+                throw new GeneratorProjectNotExistentException();
+            }*/
         }
-        if (!generatorProj.isOpen()) {
+        /*if (generatorProj.isOpen()) {
             generatorProj.open(new NullProgressMonitor());
-        }
+        }*/
```

Comment:
why was opening the configuration project removed?
In case a configuration project exists, it is open, and there is no downloaded templates jar, we now may want to raise a yes/no dialog, to ask the user to open the project before asking for downloading a new jar.

Reply:
Actually, when a CobiGen_Templates exists on the Eclipse workspace and it is closed, CobiGen can still read the templates and generate perfectly fine.

I don't really get why we should force our user to open the CobiGen_Templates project-